source/common/http/http2/codec_impl.cc

```diff
@@ -1092,13 +1092,18 @@ const ConnectionImpl::StreamImpl* ConnectionImpl::getStream(int32_t stream_id) c
   return const_cast<ConnectionImpl*>(this)->getStream(stream_id);
 }
 
-ConnectionImpl::StreamImpl* ConnectionImpl::getStream(int32_t stream_id) {
+ConnectionImpl::StreamImpl* ConnectionImpl::getStreamUnchecked(int32_t stream_id) const {
```

Comment:
I'd remove the const on the method here since we're returning a `ConnectionImpl::StreamImpl*` which is ultimately tracked in the ConnectionImpl class, and we can end up modifying in onInvalidFrame.

Meyers had a good example of this like:
```
std::string& Foo::getString() const { static std::string s = "Hello"; return s;  };

Foo myfoo;
std::string& res = myfoo.getString();
res[0] = 'J'; // static std::string s now has 'Jello' as string even though const method 🙃 
```

Reply:
Well, @adisuissa asked for the method to be `const`. Can you negotiate with him, which way to go?

Reply:
Thanks, I've missed the call to `setDetails`, otherwise it should've been a const method and a const return type.
I agree with @KBaichoo.

Reply:
Done